pipeline/pipeline.go

```diff
@@ -0,0 +1,111 @@
+package pipeline
+
+import "github.com/51Degrees/device-detection-go/v4/dd"
+
+type Pipeline struct {
```

Comment:
this corresponds to what is `DeviceDetectionOnPremiseEngine` in other languages, not `Pipeline`.